### ISSUE TITLE
Bug: Connection aborted when trying to use sublime-rest-client on Dockerized service

### ISSUE THREAD

**Issue by brupelo** [yeraydiazdiaz/sublime-rest-client#20]
@yeraydiazdiaz Hi, how you doing? Today I've had time to play around a little bit more with this one and I've catched some
issue when trying to hit some dockerized service.

Consider this mcve:

**main.py**

```
from fastapi import FastAPI

app = FastAPI()

@app.get("/")
def root():
    return {"message": "Hello"}

```

**requirements.txt**

```
fastapi
uvicorn
```

**Dockerfile**

```
FROM python:3.11.1-alpine
RUN apk add --no-cache build-base
WORKDIR /app
COPY requirements.txt .
RUN pip install --no-cache-dir -r requirements.txt
COPY . .
EXPOSE 8000
CMD ["uvicorn", "main:app", "--host", "0.0.0.0", "--port", "8000", "--workers", "2", "--reload"]

```

**requests.rest**

GET http://localhost:8000

###

If you run this service manually without docker... the request will work but if you try to hit it through Docker
such as

```
docker build -t foo .
docker run -p 8000:8000 foo
```

You'll get a crash:

```
REST Client: Error on request to http://localhost:8000

MaxRetryError("HTTPConnectionPool(host='localhost', port=8000): Max retries exceeded with url: / (Caused by ProtocolError('Connection aborted.', ConnectionResetError(10054, 'An existing connection was forcibly closed by the remote host', None, 10054, None)))")

Traceback (most recent call last):
  File "D:\software\sublimetext\sublimetext\Data\Packages\sublime-rest-client/deps\urllib3\connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "D:\software\sublimetext\sublimetext\Data\Packages\sublime-rest-client/deps\urllib3\connectionpool.py", line 449, in _make_request
    six.raise_from(e, None)
  File "<string>", line 3, in raise_from
  File "D:\software\sublimetext\sublimetext\Data\Packages\sublime-rest-client/deps\urllib3\connectionpool.py", line 444, in _make_request
    httplib_response = conn.getresponse()
  File "./python3.8/http/client.py", line 1347, in getresponse
  File "./python3.8/http/client.py", line 307, in begin
  File "./python3.8/http/client.py", line 268, in _read_status
  File "./python3.8/socket.py", line 669, in readinto
ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\software\sublimetext\sublimetext\Data\Packages\sublime-rest-client\plugin.py", line 34, in run
    self.response = client.request(self.request)
  File "D:\software\sublimetext\sublimetext\Data\Packages\sublime-rest-client\rest_client\client.py", line 21, in request
    response = http.request(
  File "D:\software\sublimetext\sublimetext\Data\Packages\sublime-rest-client/deps\urllib3\request.py", line 74, in request
    return self.request_encode_url(
  File "D:\software\sublimetext\sublimetext\Data\Packages\sublime-rest-client/deps\urllib3\request.py", line 96, in request_encode_url
    return self.urlopen(method, url, **extra_kw)
  File "D:\software\sublimetext\sublimetext\Data\Packages\sublime-rest-client/deps\urllib3\poolmanager.py", line 376, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "D:\software\sublimetext\sublimetext\Data\Packages\sublime-rest-client/deps\urllib3\connectionpool.py", line 815, in urlopen
    return self.urlopen(
  File "D:\software\sublimetext\sublimetext\Data\Packages\sublime-rest-client/deps\urllib3\connectionpool.py", line 815, in urlopen
    return self.urlopen(
  File "D:\software\sublimetext\sublimetext\Data\Packages\sublime-rest-client/deps\urllib3\connectionpool.py", line 815, in urlopen
    return self.urlopen(
  File "D:\software\sublimetext\sublimetext\Data\Packages\sublime-rest-client/deps\urllib3\connectionpool.py", line 787, in urlopen
    retries = retries.increment(
  File "D:\software\sublimetext\sublimetext\Data\Packages\sublime-rest-client/deps\urllib3\util\retry.py", line 592, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=8000): Max retries exceeded with url: / (Caused by ProtocolError('Connection aborted.', ConnectionResetError(10054, 'An existing connection was forcibly closed by the remote host', None, 10054, None)))
```

I've tried using curl directly and it works... I've tried using rest extension from vscode and it also works... but when i try with sublime-rest-client it crashes. The funny thing is, if I run uvicorn manually without docker sublime-rest-client will work as well.

Any idea what's going on?

**Comment by brupelo** [comment]
Trying to hit `GET http://127.0.0.1:8000/` will work with the dockerized service though. In vscode's extension both `127.0.0.1` and `localhost` working with the dockerized version

**Comment by yeraydiazdiaz** [comment]
Hey @brupelo, I just tried this on my Mac and it seems to work fine both with `http://localhost:8000` and `http://127.0.0.1:8000`. I see you're running Windows so I'm not able to fully reproduce your problem.

My guess is that there's some name resolution conflict between ST and the OS, you try opening the ST console selecting View > Show console and typing:

```
import socket
socket.gethostbyname('localhost')
```

In my Mac it resolves to `127.0.0.1`.

I also tried `socket.gethostbyaddr('127.0.0.1')` which returned `('localhost', ['1.0.0.127.in-addr.arpa'], ['127.0.0.1'])`.

Proving the name resolution works fine in ST which I suspect it doesn't in your system. Have a go a it and let me know what you find.
